app/controllers/offers_controller.rb

```diff
@@ -0,0 +1,108 @@
+class OffersController < ApplicationController
+  before_action :set_trade, only: [:new, :show, :create, :edit, :update, :accept]
```

Comment:
If you are using a `before_action` on all the actions in a controller I believe you can leave off the `only`

`before_action :set_trade`